app/scripts/controllers/transactions.js

```diff
@@ -382,9 +382,12 @@ module.exports = class TransactionController extends EventEmitter {
     this._setTxStatus(txId, 'confirmed')
   }
 
-  setTxStatusFailed (txId, reason) {
+  setTxStatusFailed (txId, err) {
     const txMeta = this.getTx(txId)
-    txMeta.err = reason
+    txMeta.err = {
```

Comment:
`err.toString` is this the hole error? because messages will be shown in the ui
so would prefer this not to be the message if thats the case.